### ISSUE TITLE
Logical Operators in story file?

### ISSUE THREAD

**Issue by ctrado18** [RasaHQ/rasa_core#1069]
Hey,

is it possible to write stories with operators like NOT. I want to write a story where action is executed if not a specific intent is given without writing for every intent except the wanted one the stories. I mean:

```

* NOT Intent_Wanted
    - utter_fallback
```

**Comment by Ghostvv** [comment]
At this moment exclusion operator is not supported, as it is not clear how to train a model in this case

**Comment by akelad** [comment]
Yeah so you would have to handle it with adding an `OR` statement for all other intents

**Comment by ctrado18** [comment]
OR and AND will work like:

```
* Intent_Wanted OR * other_intent

```

It would be nice to have context based fallback such that based on prevoius intent or action like asking_for_location. If user gives then not required or requested intent than do fallback.

See:

https://forum.rasa.com/t/infer-intent-from-extracted-entity-context/1370

**Comment by akelad** [comment]
what do you mean AND? that doesn't exist. But or should look like this:
```
* Intent_Wanted OR other_intent
```
As for this context based fallback, you can write a story for that, if the user gives you any intent other than something you wanted, send a different reply. I think the post you linked is addressing something else

**Comment by ctrado18** [comment]
You can need this to do out of context fallbacks without writing exceptions for every intent. This is whta the link is about. But it would be nice to handle this in more rigourus way like here: https://github.com/mrbot-ai/rasa-addons#swap-intents 

> Some intents are hard to catch. For example when the user is asked to fill arbitrary data such as a date or a proper noun. The following rule swaps any intent caught after utter_when_do_you_want_a_wake_up_call with enter_data unless...

```

intent_substitutions:
  - after: utter_when_do_you_want_a_wake_up_call
    intent: enter_data
    unless: frustration|cancel|speak_to_human
```

**Comment by akelad** [comment]
Feel free to use this rasa-addons package, currently we're not going to add something like this to Rasa core. But when it comes to slot filling we'll be merging a new version of forms where the intent isn't payed attention to as much
